### DISCUSSION TITLE
Coin-holder 'protection' when a creator wants to sell their coin.

### DISCUSSION THREAD

**Discussion by 0xNordian** [deso-protocol/dips#176] [Medium CIPs]
**name:** Coin-holder 'protection' when a creator sells their coin.
**about:** Temporary sell request standby + coin-holder communication when creators want to sell their own coin.
**title:** N/A
**labels:** Creator Coin (CC), Coin-Holders, CC sell communication, Rug-Pull 
**assignees:** N/A

### **One-Line Description**
Create a core rule that gives coin-holders a fair chance to decide what they want to do when a creator wants to sell their own coin.

### **Explanation and Motivation**
Creators access to liquidity is fundamental for Deso success. However, if not done correctly and access to this liquidity, it's done with the wrong methods or lack of transparency, it will damage the trust in the protocol/nodes/users.

Selling your coin it's not a bad thing perse, but potentially taking the advantage to sell your coin without giving a fair opportunity to coin-holders to step out if wanted might be counterproductive for Deso.

**_High-level proposed action:_** The CC sell order will be on hold for 'X' time; a notification is automatically sent to coin holders about the CC selling request. After X period is due, the order gets executed. 

**_Detailed proposed action:_**
0. A creator request to sell X amount of their coin.

1. The CC selling request is on hold for X amount of time (hours, days, etc. The community should agree on the period *).

2. An automatic notification will be sent to each CC holder, warning them that a sell of X amount of CC $DESO will occur by X date. **[MANDATORY]**

3. The seller can include a message to be sent to coin holders to justify the action (this data could be input when the seller triggers the selling action) **[OPTIONAL]**

4. Both messages will give coin holders the proper time to analyze the selling reasons and decide if they want to keep the position or not.

5. Once the period is reached, the CC selling request gets executed.

Some benefits:
- Improve the trust of coin-holders.
- Increase investment in creator coins due to feeling of 'security.'
- Opportunity for creators to justify a selling order to access liquidity.
- Increase of trust in the Deso Protocol.

Trade-off:

- Delay for creators to access liquidity.
- Amount of liquidity access for creators if major coin-holder sell-off happens. 

Variations:
*Creators can configure a 'pre-advice' time when selling their own coin in their accounts. This could be a visible piece of information (as the Founder Reward), so it could become something flexible yet strategical for each creator to decide and each 'investor' to know in advance before investing. (Potential complication on how to store the period per transaction data).

**Comment by DesoTrollz** [comment]
Great idea! Appreciate this being addressed. We weren't too sure if we were overexposed with ours and putting off new investors so it's good to see issues being raised 🙌

**Comment by DegenDudes** [comment]
Sounds centralized, that's not how cryptos work. Imagine if BTC worked like that, entities could buy into it and keep Satoshi from selling? This could be used maliciously for the wealthy to mute the powers of the poorer people to even being able to sell their own coin. This is a horrible idea.Can you control who can buy your coin then? And your holders have to have consensus?

**Reply by 0xNordian** [reply]
I respect if you disagree. I'm not sure to understand why it is centralized, in your opinion. I don't think it's comparable with the BTC market; the way the bonding curve works and the fact that you don't need to sell your coins to someone (just need to burn them) doesn't create a trusted environment that is fair for investments, IMHO. 

Besides the criticism, that is more than welcome. What is your opinion regarding the creator coins? Do you think we have an issue with the trust in them? Do we have a problem with the rug pulls? Can a selling of a creator coin be considered 'inside trading'? Do you have any proposal to improve the current configuration?

Thanks for your feedback.

**Reply by DegenDudes** [reply]
Still would you want your government being able for instance to delegate your own coin sale? If your cool with that then ok...

**Reply by DegenDudes** [reply]
I will work on better feedback in how the threat could be mitigated for manipulation, I guess if it could be chosen as a setting maybe upon wallet creation that cannot be undone. Register your wallet as an "investment" CC specifically.

**Reply by 0xNordian** [reply]
> Still would you want your government being able for instance to delegate your own coin sale? If your cool with that then ok...

I have to clarify that English is not my primary language, so maybe I misexpressed. Please, if you can let me know in which part of the text I'm proposing, the delegation of the selling of a creator coin will be great so I can amend it if necessary.

The creator coin sell order doesn't need approval; the proposal only intends to open a window for coin-holders to evaluate if they would like to step out of a position. The sale of the creator coin will always happen without any delegation or authorization.

**Reply by DegenDudes** [reply]
It isn't centralized per say, but it allows a direct gateway for centralized manipulation. If this is to be a core feature, then you should allow the "legacy" wallet structure to remain intact and offer this as an option. I think having multiple CC structures for what you plan on using deso would be a good thing.

You could simply do this by having "personal" or "business" and the business would contain your proposed structure.

**Reply by DegenDudes** [reply]
> 

Ok I misread that. My error. 🤣

**Reply by DegenDudes** [reply]
> It isn't centralized per say, but it allows a direct gateway for centralized manipulation. If this is to be a core feature, then you should allow the "legacy" wallet structure to remain intact and offer this as an option. I think having multiple CC structures for what you plan on using deso would be a good thing.
> 
> You could simply do this by having "personal" or "business" and the business would contain your proposed structure.

Then for CC settings a toggle for allow or disallow business or personal accounts to purchase your CC.

**Comment by starykolarz** [comment]
I see a simple workarund, or in fact 2, users can use to make such a mechanism practically obsolete:
1. create one or more legitimately looking accounts and buy your "main" coin with those
2. cross-buy your own coin with a friend / friends who you trust

The legit users will stick to buying their own coin, informing coinholders before own sale because they feel it's right - marginal impact on them or their coinholders.
The ones that will want to scam, will anyway, using one of the above mechanisms, or any other they think of.... and last months have shown that scammers are creative.

**Reply by 0xNordian** [reply]
You are not wrong. In fact, that is a practice used today. The proposal only intends to give more 'protection' mechanisms to coin-holders. I think in the scenario you are using as a example it will be easy to identify this practice since if you see a coin-holder or a series of them hold far more coins than the creator probably it's something at least will make people think twice before investing. 

Of course, that can happen if someone like whoami invests so much money in you that surpasses your own investment. However, in that case, we will be talking about a well-known user so the 'trust' 'clout' 'trustworthiness' of a user comes into play.

I don't think there is a way to avoid rug pulls 100% but at this stage feels simply like the 'far west'. The only mechanism that exists for Coin holders not to get rug pulled is praying 😅.

**Reply by starykolarz** [reply]
Another idea - why not borrow from INCs? That would mean that creator cannot buy own coin - so no chance to rug-pull, or sell own coin.
Of course point 1 from my initial comment still is possible - but that's same as insider trading stocks, and while cannot be prosecuted on blockchain, it can be stigmatized in a way visible for others.

**Reply by arhebbar** [reply]
Insider Trading is a given here and I almost feel like the only way to prevent it is to frame a set of rules when the DAOs are in place and have every Creator Coin become or sign-up to one type of DAO with rules pre-defined before inviting investors. Almost like a badge.  

Anyone who buys into the coin of a Creator is joining the DAO created by the Creator and each Creator can choose what type of DAO he wants to create his coin under - The "No Buy DAO", the "NOTIFY Sells DAO" or the "I won't buy more coins after first buy" DAO.  But, my opposition to the original proposal is to bringing this into core since anything we do will not meet all the usecases that DeSo CCs will need to be used for and may be limiting.

**Comment by arhebbar** [comment]
> Selling your coin it's not a bad thing perse, but potentially taking the advantage to sell your coin without giving a fair opportunity to coin-holders to step out if wanted might be counterproductive for Deso.

While this is a problem, I don't think preventing people from selling their coin is fair. The assumption here is the coin dynamics are as it is today. What if I haven't bought all my coins in the beginning and I feel like one of my largest coin holders is exiting the platform. Why should I be prevented from selling all my coins when my largest coin holders can?

The way I look at it, if I do rug-pull, the data is on the blockchain and soon, there will be tags on rug-pullers that will warn future investors. So, if someone does want to rug-pull, good luck to him getting any investment on any DESO platform. 

I would look at a solution similar to what Brootle suggested in the profile image DIP. Use the "UserProfileExtraData" by tagging Rug-Pullers so that future investors get that information before they invest or engage.

**Reply by 0xNordian** [reply]
>I don't think preventing people from selling their coin is fair.

The proposal doesn't intend to prevent people from selling their own coins. I only open a window for coin-holders to decide if they want to step out before the coin gets sold. 

>Why should I be prevented from selling all my coins when my largest coin holders can?

I insist the proposal doesn't say in any way that you will not be able to sell your coin. Will only delay the sale order to allow proper notification to the coin holder, after which the sell order will be executed.

>Use the "UserProfileExtraData" by tagging Rug-Pullers

I get the point, but I'm afraid I have to disagree with it. Not all the creator coin sales are rug pulls; to set by default a rug pull label to anyone that sells their coin might not be fair. In my proposal, the creator has the opportunity to explain the reason to sell their coin, so coin-holders can decide what to do. After that, they sale the coin and IMO cannot be considered a rug pull since everybody was appropriately informed.

Thanks a lot for taking the time to share your thoughts. 🤘

**Reply by arhebbar** [reply]
Like I said, I am not in favor of this proposal in its current state. Explaining my comment with two examples:

1. Imagine a user who bought most of his 5 coins at 100$ (I did buy 5 of my coins after it hit 150$ btw) after other investors had already bought 10 coins to bring it up to 100$. By informing investors about his intent to sell, in all likelihood, the Creator will get rug-pulled by his early investors. Informing investors that you are selling is as good as encouraging your big investors to exit leading to a loss for the Creator on his own coin.

2. Over time, I look at Creator Coin investments as investing in people you work with and/or know irl or thru collabs. In those scenarios, there will be trust automatically and forcing Creators to explain any sale (for any purpose) creates an uncomfortable situation for the Creator and possibly the investor. Like @mashelenn said, this has to be voluntary and not forced. Also, how do you decide X is going to be a bigger question.

PS: I was referring to those who sell all their coins as "Rug-Pullers" because that's what they are even if they have their reasons. I know it has a negative connotation. But, if you take the literal term, they most likely pulled the rug under their investor's feet. 👎

**Reply by arhebbar** [reply]
One more thing I thought about after seeing a comment above from @PineappleClout. What if a hostile investor (some one the Creator doesn't like irl or someone you have a disagreement with has majority of your coin), the fact that the investor can sell at any time, but the Creator can't is an unfair system in my mind.

**Reply by mashelenn** [reply]
There is another version of events.
A person decides to sell his coin.
He sends messages to his investors.
As a result decides not to sell - changes his mind to do so for some reason.
What to do in this case?)
At least the person should have a choice: to report or not to report the sale of his coin.

**Comment by mashelenn** [comment]
I don't think alerting investors to the sale of the coin can take place at DeSo.
It would create tension and guilt on both sides.
I think reputation is important in any space. Whether it's the Internet or real life, it doesn't matter.
Information that a person has previously sold their coin is no secret.
It is up to the investor to decide whether or not to invest in that person's coin.
I think this is insider information that only the owner of the coin himself can possess. The decision of whether or not to report what he is going to sell should be the person's own responsibility.
Otherwise, it's leverage from the system.

**Reply by 0xNordian** [reply]
> It would create tension and guilt on both sides.

I get your point. However, I don't think this will create more pressure than selling without any feasible way to inform coin-holders properly.

>I think this is insider information that only the owner of the coin himself can possess. Whether or not to report what he is going to sell should be the person's responsibility.

Agree; that is why I mentioned that one variant to the proposal could be to make this option optional per creator at the end of the proposal. However, there are no tools or features to communicate efficiently with your coin-holders at this stage, and not even a simple bulk message to your coin holders.

Thanks so much for your feedback and time! 🤘

**Comment by 0xNordian** [comment]
Well, I guess today we could have tested this proposal 😅

**Comment by oO** [comment]
Is the problem more with the overly aggressive bonding curve of the coins rather than the whole buy/sell/FR? the only reason this is a problem in the first place is because the bonding curve of the coins makes them extremely volatile. Although, to be fair, anything with a minuscule liquidity pool like most CC is going to have enormous volatility. What do you think would happen to TSLA if Elon Musk tried to sell his entire stake?

**Comment by 1R053** [comment]
I proposed a different long-term solution for the problem here:
https://github.com/deso-protocol/dips/discussions/191

Essentially, a creator should be able to choose the type of bonding curve he wants his followers / investors to use. Typically, a quadratic bonding curve is not what I would expect creators to choose that are looking for long-term engagement.
